### DISCUSSION TITLE
Prometheus bearer token

### DISCUSSION THREAD

**Discussion by pat-s** [containrrr/watchtower#904] [Q&A]
I am new to Prometheus and Grafana and tried to enabling the "Metrics" endpoint.

I cannot find how to create the required `bearer_token` for Prometheus as shown in the `docker-compose` [example](https://github.com/containrrr/watchtower/blob/3b60afe55378bdb36554e9e9dff2eed962c184a2/prometheus/prometheus.yml#L5).

Could someone help me here?

**Comment by simonaronsson** [comment]
Sure! I think this is the argument you're looking for (https://containrrr.dev/watchtower/arguments/#http_api_token)

> Sets an authentication token to HTTP API requests.

```
            Argument: --http-api-token
Environment Variable: WATCHTOWER_HTTP_API_TOKEN
                Type: String
             Default: -
```

**Reply by piksel** [reply]
Note that the token is arbitrary and can be set to anything you like. It just needs to be the same as the string passed to watchtower.

**Reply by pat-s** [reply]
Thanks!

I think I was not specific in my OP - I was aware how to set it in `watchtower` but I could not figure out how to create a `bearer_token` in Prometheus.

I am aware that this might be out of scope here but if someone has already done this, sharing how to do it / link would be highly appreciated 🙏

**Reply by piksel** [reply]
Yeah, that was what my comment referred to. You don't create anything specifically *for* Prometheus, it's just a configuration that you add, and the value can be anything you like.  You can use 
```sh
dd if=/dev/random count=1 status=none | sha256sum
```
if you are not feeling creative 😁.

The line that you linked is just where you *tell* Prometheus what token to use. So just put any string you want in your own `/etc/prometheus/prometheus.yml` and pass the same string to watchtower.

**Reply by pat-s** [reply]
Ahh 🤦 Sorry, this was not clear to me 😅 

So now I've got the `docker-compose` stack running and trying to connect to the Grafana port.

On the Grafana dashboard I get asked for login credentials - do I need to configure this externally in the Grafana YML files?
